login_all_company/static/src/js/company_service.esm.js

```diff
@@ -0,0 +1,20 @@
+/** @odoo-module **/
+
+import {companyService} from "@web/webclient/company_service";
+import {patch} from "web.utils";
+import {session} from "@web/session";
+
+patch(companyService, "login_all_company/static/src/js/company_service.esm.js", {
+    start(env, {user, router, cookie}) {
+        if (!cookie.current.cids && !router.current.hash.cids) {
+            var companies = [session.company_id];
+            _.forEach(session.user_companies.allowed_companies, function (company) {
+                if (company.id !== session.company_id) {
+                    companies.push(company.id);
+                }
+            });
+            cookie.setCookie("cids", companies.join(","));
+        }
+        return this._super(env, {user, router, cookie});
```

Comment:
use this._super(...arguments) this will ensure that all arguments are passed if some are not used by your function

Reply:
If you don't mind, I'd prefer not to change this, mostly because we've both already tested it as is, but also because this matches the signature of the method in Odoo 15:

https://github.com/odoo/odoo/blob/670012a04f9d849ca8f0a42796ae47ecf138eaf8/addons/web/static/src/webclient/company_service.js#L34

Reply:
Eh... even as I write that, I see the value in not blocking off avenues of extension of the company service that I cannot imagine.

Reply:
Thanks, I wouldn't have thought of that; I don't write much Javascript.

Reply:
Don't worry, that's part of the review process :smile: